### ISSUE TITLE
Rationalizing existing NUNit test suites for the compiler

### ISSUE THREAD

**Issue by cartermp** [dotnet/fsharp#7076]
We currently have two separate test NUnit test projects for the compiler:

FSharp.Compiler.UnitTests: https://github.com/dotnet/fsharp/tree/master/tests/FSharp.Compiler.UnitTests

FSharpSuite.Tests: https://github.com/dotnet/fsharp/blob/master/tests/fsharp/FSharpSuite.Tests.fsproj

I don't see too many meaningful differences between the two. Should these be merged into one? Or should better names be picked if there's a meaningful difference? Currently I don't know where I should stick a new compiler unit test.

**Comment by cartermp** [comment]
Related to #7075 - arguably this is a prerequisite.

**Comment by cartermp** [comment]
@TIHan any preferences?

**Comment by forki** [comment]
Ok. Which one should be the survivor?

**Comment by cartermp** [comment]
The one with the better name 😄

**Comment by TIHan** [comment]
Originally I started doing tests in `Compiler.UnitTests`, but I had to move it out and into `FSharpSuite` to take advantage of its rules to figure out where the paths to certain things are. After looking at it again, I think we could move some of them back (we don't have very many).

Ideally, it would all live in `Compiler.UnitTests`.

Also, we should start looking into how to be like Roslyn when it comes to testing this stuff.
